tests/net/route/src/main.c

```diff
@@ -355,167 +336,129 @@ static bool populate_nbr_cache(void)
 	recipient = my_iface;
 
 	if (!net_test_send_ns(peer_iface, &peer_addr)) {
-		return false;
+		zassert_true(0, NULL);
 	}
 
 	nbr = net_ipv6_nbr_add(net_if_get_default(),
 			       &peer_addr,
 			       &net_route_data_peer.ll_addr,
 			       false,
 			       NET_IPV6_NBR_STATE_REACHABLE);
-	if (!nbr) {
-		TC_ERROR("Cannot add peer to neighbor cache\n");
-		return false;
-	}
+	zassert_not_null(nbr, "Cannot add peer to neighbor cache");
 
 	k_sem_take(&wait_data, WAIT_TIME);
 
 	feed_data = false;
 
 	if (data_failure) {
 		data_failure = false;
-		return false;
+		zassert_true(0, NULL);
```

Comment:
please get rid of all these zassert_true(0, NULL) and replace with a direct test and a string explaining what the failure means

Reply:
Sure. In some cases inside if loop, new value has been assigned to variable. In that case I can add failure message. Rest of cases I will replace with direct test and a string with failure message.

Reply:
Fixed